### DISCUSSION TITLE
Is there any way to load PMTiles?

### DISCUSSION THREAD

**Discussion by imakihi** [TerriaJS/terriajs#7086] [Q&A] [answered]
Hi Terria community!
I was wondering there is a way to load PMTiles vactor data. I tried to load my PMTiles file as MVT but no success...
It would be winderful if someone knows if I could use PMTiles with TerriaJS.

Thank you!

**Comment by techmavengeospatial** [comment]
LOAD YOUR PMTILES INTIO MARTIN https://github.com/maplibre/martin or extract as folder of tiles (XYZ or TMS) 
served from AZURE BLOB Storage or S3 Bucket or any hosting account.
Terriajs does not support PMTILES

https://docs.protomaps.com/pmtiles/cli

We have App to convert between map tile formats
https://portfolio.techmaven.net/apps/tile-utilities/

**Reply by imakihi** [reply]
@techmavengeospatial 
Thank you for your input. The reson to use PMTiles for me was that I didn't want to use any server...
I also want to keep my data as vector, not raster....
Best.

**Comment by tephenavies** [comment]
Hi @imakihi. Using the "mvt" type should work with pmtiles, but i's possible that your pmtiles archive is a newer version than is supported currently in terriajs. Could you please check if http://ci.terria.io/upgrade-protomaps is able to visualise your pmtiles?

**Reply by imakihi** [reply]
Hi @steve9164. Thank you for responding this.
I tried my PMTiles with your app and it worked!! But my version still couldn't show the same layer yet.
What I need to do to show my PMTiles with my TerriaJS?
I think my version is 8.3.4.

Thank you for your help!!!
![image](https://github.com/TerriaJS/terriajs/assets/4824436/ecc87cb1-9757-4306-9f35-08f299a5b74c)

**Reply by tephenavies** [reply]
The app I sent is running a build from an unmerged PR to TerriaJS https://github.com/TerriaJS/terriajs/pull/6725. Unfortunately that PR is stalled on some details. I'll push through another PR with just an upgrade to our vector mapping library that supports this updated version of pmtiles archives.

**Reply by imakihi** [reply, marked as answer]
@steve9164 
That would be super super cool. I look forward to useing PMTiles on our Terria Maps.
Best.
